integration-test/src/main/scala/com/cloudera/livy/test/framework/LivyRestClient.scala

```diff
@@ -0,0 +1,242 @@
+        newStmt.id
+      }
+
+      @tailrec
+      final def result(): Either[String, StatementError] = {
+        val r = httpClient.prepareGet(s"$url/statements/$stmtId")
+          .execute()
+          .get()
+        assertStatusCode(r, HttpServletResponse.SC_OK)
+
+        val newStmt = mapper.readValue(r.getResponseBodyAsStream, classOf[StatementResult])
+        if (newStmt.state == "running") {
+          Thread.sleep(1000)
+          result()
+        } else {
+          val output = newStmt.output
+          output("status") match {
+            case "ok" =>
+              val data = output("data").asInstanceOf[Map[String, Any]]
+              Left(data("text/plain").asInstanceOf[String])
+            case "error" =>
+              Right(mapper.convertValue(output, classOf[StatementError]))
+            case status =>
+              throw new Exception(s"Unknown statement status: $status")
+          }
+        }
+      }
+
+      def verifyResult(expectedRegex: String): Unit = {
+        result() match {
+          case Left(result) =>
+            if (expectedRegex != null) {
+              matchStrings(result, expectedRegex)
+            }
+          case Right(error) =>
+            assert(false, s"Got error from statement $code: ${error.evalue}")
+        }
+      }
+
+      def verifyError(
+          ename: String = null, evalue: String = null, stackTrace: String = null): Unit = {
+        result() match {
+          case Left(result) =>
+            assert(false, s"Statement `$code` expected to fail, but succeeded.")
+          case Right(error) =>
+            val remoteStack = Option(error.stackTrace).getOrElse(Nil).mkString("\n")
+            Seq(error.ename -> ename, error.evalue -> evalue, remoteStack -> stackTrace)
+              .foreach {
+                case (actual, expected) =>
+                  if (expected != null) {
+                    matchStrings(actual, expected)
+                  }
+              }
+        }
+      }
+
+      private def matchStrings(actual: String, expected: String): Unit = {
+        val regex = Pattern.compile(expected, Pattern.DOTALL)
+        assert(regex.matcher(actual).matches(), s"$actual did not match regex $expected")
+      }
+    }
+
+    def run(code: String): Statement = { new Statement(code) }
+
+    def verifySessionIdle(): Unit = {
+      verifySessionState(SessionState.Idle())
+    }
+  }
+
+  def startBatch(
+      file: String,
+      className: Option[String],
+      args: List[String],
+      sparkConf: Map[String, String]): BatchSession = {
+    val r = new CreateBatchRequest()
+    r.file = file
+    r.className = className
+    r.args = args
+    r.conf = Map("spark.yarn.maxAppAttempts" -> "1") ++ sparkConf
+
+    val id = start(BATCH_TYPE, mapper.writeValueAsString(r))
+    new BatchSession(id)
+  }
+
+  def startSession(kind: Kind, sparkConf: Map[String, String]): InteractiveSession = {
```

Comment:
These are the factory methods for InteractiveSession. I can move them to `object InteractiveSession`, but not `class InteractiveSession` because InteractiveSession's constructor requires an id. I don't want to declare another companion object here so I would prefer to keep them in LivyRestClient.